src/uu/stty/src/stty.rs

```diff
@@ -203,8 +209,35 @@ fn stty(opts: &Options) -> UResult<()> {
     let mut termios = tcgetattr(opts.file.as_fd()).expect("Could not get terminal attributes");
 
     if let Some(settings) = &opts.settings {
-        for setting in settings {
-            if let ControlFlow::Break(false) = apply_setting(&mut termios, setting) {
+        let mut settings_iter = settings.iter();
+        while let Some(setting) = settings_iter.next() {
+            if let Some(char_index) = is_control_char(setting) {
+                let Some(new_cc) = settings_iter.next() else {
+                    return Err(USimpleError::new(
+                        1,
+                        format!("missing argument to '{setting}'"),
+                    ));
+                };
+                match apply_char_mapping(&mut termios, char_index, new_cc) {
+                    Ok(_) => {}
+                    Err(e) => match e {
+                        ControlCharMappingError::IntOutOfRange => {
+                            return Err(USimpleError::new(
+                                1,
+                                format!(
+                                    "invalid integer argument: '{new_cc}': Numerical result out of range"
+                                ),
+                            ));
+                        }
+                        ControlCharMappingError::MultipleChars => {
+                            return Err(USimpleError::new(
+                                1,
+                                format!("invalid integer argument: '{new_cc}'"),
+                            ));
+                        }
+                    },
+                }
```

Comment:
As you only handle the error case, you can simplify it with `map_err`:
```suggestion
                apply_char_mapping(&mut termios, char_index, new_cc).map_err(|e| {
                    let message = match e {
                        ControlCharMappingError::IntOutOfRange => format!(
                            "invalid integer argument: '{new_cc}': Numerical result out of range"
                        ),
                        ControlCharMappingError::MultipleChars => {
                            format!("invalid integer argument: '{new_cc}'")
                        }
                    };
                    USimpleError::new(1, message)
                })?;
```